hare/flows_test.go

```diff
@@ -277,116 +232,6 @@ func (m *mockClock) advanceLayer() {
 	m.currentLayer = m.currentLayer.Add(1)
 }
 
-type SharedRoundClock struct {
```

Comment:
What was the purpose of this clock?

Reply:
no idea. i don't see it valuable as it cause the test not to exercise the actual code.

Reply:
From my understanding it was used to

Reply:
yes, but thats just not realistic testing. i dont know why it doesnt used fixed round durations. it also make test failures difficult to reproduce. just cannot see the value here, hence the removal